### PR TITLE
feat: Configurable CICS translator

### PR DESCRIPTION
**Pull request by ap891843**
Configurable CICS translator

Signed-off-by: ap891843 <aman.prashant@broadcom.com>

### CODE REVIEW

File: clients/cobol-lsp-vscode-extension/package.json [hunk 1]
```diff
@@ -443,6 +443,11 @@
                     ],
                     "uniqueItems": true
                 },
+                "cobol-lsp.cics.translator": {
+                    "type": "boolean",
+                    "default": "true",
+                    "description": "CICS translator"
```
Comment (ap891843) [inline clients/cobol-lsp-vscode-extension/package.json:449]: @VitGottwald @zeibura @zimlu02  Any suggestion for this?

File: server/engine/src/main/java/org/eclipse/lsp/cobol/core/model/tree/logic/CICSTranslatorProcessor.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  public void accept(CICSTranslatorNode cicsTranslatorNode, ProcessingContext processingContext) {
+    if (!analysisConfig.isCicsTranslatorEnabled()) {
+      processingContext
+          .getErrors()
+          .add(
+              SyntaxError.syntaxError()
+                  .locality(cicsTranslatorNode.getLocality())
+                  .severity(ErrorSeverity.ERROR)
+                  .errorSource(ErrorSource.PARSING)
+                  .suggestion("Enable CICS translator")
```
Comment (ap891843) [inline server/engine/src/main/java/org/eclipse/lsp/cobol/core/model/tree/logic/CICSTranslatorProcessor.java:47]: @VitGottwald  @zeibura @zimlu02 Any suggestion for the error message when cics is disabled but we find CICS translator options

File: server/engine/src/main/java/org/eclipse/lsp/cobol/core/engine/dialects/cicsTranslator/CICSTranslatorDialect.java [hunk 3]
```diff
@@ -1,71 +0,0 @@
-/*
```
Comment (ishche) [inline server/engine/src/main/java/org/eclipse/lsp/cobol/core/engine/dialects/cicsTranslator/CICSTranslatorDialect.java:1]: 💯

**Review by ishche — APPROVED** [review]

**Review by Nurkambay — APPROVED** [review]

File: server/engine/src/main/java/org/eclipse/lsp/cobol/core/model/tree/logic/CICSTranslatorProcessor.java (hunk 2 above)
Comment (zimlu02) [reply server/engine/src/main/java/org/eclipse/lsp/cobol/core/model/tree/logic/CICSTranslatorProcessor.java:47]: maybe "Enable CICS translator in the settings" ?

File: clients/cobol-lsp-vscode-extension/package.json (hunk 1 above)
Comment (zimlu02) [reply clients/cobol-lsp-vscode-extension/package.json:449]: "Allow CICS translator" ?

Comment (zeibura) [reply clients/cobol-lsp-vscode-extension/package.json:449]: Enable CICS translator. BTW what is a CICS translator?

File: server/engine/src/main/java/org/eclipse/lsp/cobol/core/model/tree/logic/CICSTranslatorProcessor.java (hunk 2 above)
Comment (zeibura) [reply server/engine/src/main/java/org/eclipse/lsp/cobol/core/model/tree/logic/CICSTranslatorProcessor.java:47]: CICS is disabled where?

Comment (ap891843) [reply server/engine/src/main/java/org/eclipse/lsp/cobol/core/model/tree/logic/CICSTranslatorProcessor.java:47]: To enable CICS compilation in the cobol program, **CICS translator** compiler directives were used. But with the latest cobol version, this support is inbuilt and doesn't require these directives. Some customers want to support these statements while others want to throw errors on them.

File: clients/cobol-lsp-vscode-extension/package.json (hunk 1 above)
Comment (ap891843) [reply clients/cobol-lsp-vscode-extension/package.json]: Updated to "Enable CICS translator"

File: server/engine/src/main/java/org/eclipse/lsp/cobol/core/model/tree/logic/CICSTranslatorProcessor.java (hunk 2 above)
Comment (ap891843) [reply server/engine/src/main/java/org/eclipse/lsp/cobol/core/model/tree/logic/CICSTranslatorProcessor.java]: Updated to - "Enable CICS translator in the settings"

Comment (ap891843) [reply server/engine/src/main/java/org/eclipse/lsp/cobol/core/model/tree/logic/CICSTranslatorProcessor.java]: image

File: clients/cobol-lsp-vscode-extension/package.json (hunk 1 above)
Comment (ap891843) [reply clients/cobol-lsp-vscode-extension/package.json]: image

File: server/engine/src/main/java/org/eclipse/lsp/cobol/core/model/tree/logic/CICSTranslatorProcessor.java (hunk 2 above)
Comment (ap891843) [reply server/engine/src/main/java/org/eclipse/lsp/cobol/core/model/tree/logic/CICSTranslatorProcessor.java]: Yes. If we disable the CICS translator from the extension settings, we start to show an error, if the program used the CICS translator within the cobol code. 
By default, it is set to true. 
Basically (as I understand), the CICS translator depends on the cobol compiler. If a company/organization is using an old cobol compiler they would need to have these to support CICS.
While the newer compiler doesn't need these and we need to show errors on those statements.

Comment (zeibura) [reply server/engine/src/main/java/org/eclipse/lsp/cobol/core/model/tree/logic/CICSTranslatorProcessor.java]: So I'm not sure "translator" is the best term to use in public facing texts. How about "Enable CICS support"?

**Review by zeibura — APPROVED** [review]

**Pull request merged**